jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -3029,6 +3015,72 @@ public MethodHandle run() {
 	return valueMethod;
 }
 
+private MetadataCache getMetadataCache() {
+	if (metadataCache == null) {
+		metadataCacheOffset = getFieldOffset("metadataCache"); //$NON-NLS-1$
+		writeFieldValue(metadataCacheOffset, new MetadataCache());
+	}
+	return metadataCache;
+}
+
+private String cacheSimpleName(String simpleName) {
+	MetadataCache cache = getMetadataCache();
+
+	if (cache.cachedSimpleName == null || cache.cachedSimpleName.get() == null) {
+		MetadataCache.cachedSimpleNameOffset = getFieldOffset(
+				MetadataCache.class, "cachedSimpleName", MetadataCache.cachedSimpleNameOffset); //$NON-NLS-1$
+
+		writeFieldValue(cache, MetadataCache.cachedSimpleNameOffset, new SoftReference<>(simpleName));
+	}
+
+	return simpleName;
+}
+
+private String cacheCanonicalName(String canonicalName) {
+	MetadataCache cache = getMetadataCache();
+
+	if (cache.cachedCanonicalName == null || cache.cachedCanonicalName.get() == null) {
+		MetadataCache.cachedCanonicalNameOffset = getFieldOffset(
+				MetadataCache.class, "cachedCanonicalName", MetadataCache.cachedCanonicalNameOffset); //$NON-NLS-1$
```

Comment:
```suggestion
		if (-1 == MetadataCache.cachedCanonicalNameOffset) {
			MetadataCache.cachedCanonicalNameOffset = getFieldOffset(
					MetadataCache.class, "cachedCanonicalName", MetadataCache.cachedCanonicalNameOffset); //$NON-NLS-1$
		}
```

Reply:
As discussed, `getFieldOffset` has this check already, so I will leave it as is.